### PR TITLE
[FIX] Fixed Initialization Handling

### PR DESCRIPTION
**Pull request by renesansz**
Pull Request Changes:
* Removed deprecated method `setDebug`
* (Android) Fixed plugin initialization and session management.

### CODE REVIEW

File: src/android/io/branch/BranchSDK.java [hunk 1]
```diff
@@ -221,16 +204,18 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
...
 
         Uri data = activity.getIntent().getData();
+
         if (data != null && data.isHierarchical()) {
             this.deepLinkUrl = data.toString();
         }
 
-        this.instance = Branch.getAutoInstance(this.activity.getApplicationContext());
-
+        this.instance = Branch.getInstance(this.activity.getApplicationContext());
```
Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: Please leave as getAutoInstance

File: src/android/io/branch/BranchSDK.java [hunk 2]
```diff
@@ -41,44 +41,32 @@
...
 
     /**
-     * Called when the activity will start interacting with the user.
-     *
-     * @param multitasking A {@link boolean} flag indicating if multitasking is turned on for app
+     * Called after plugin construction and fields have been initialized.
      */
     @Override
-    public void onResume(boolean multitasking) {
+    protected void pluginInitialize() {
```
Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: when does this get called?

Comment (renesansz) [inline src/android/io/branch/BranchSDK.java]: According to Cordova: "Called after plugin construction and fields have been initialized."

File: src/android/io/branch/BranchSDK.java (hunk 1 above)
Comment (renesansz) [inline src/android/io/branch/BranchSDK.java]: @aaustin updated PR, should be using `getAutoInstance` instead of `getInstance`.

**Comment by aaustin** [comment]
Can you fix the conflicts and merge in?

**Comment by renesansz** [comment]
@aaustin fixed merge conflicts

**Pull request merged**